crates/amf0/src/de/mod.rs

```diff
@@ -77,21 +77,24 @@ where
     where
         V: serde::de::Visitor<'de>,
     {
-        self.deserialize_i64(visitor)
+        let value = self.decode_number()?;
```

Comment:
do `peek_marker` and if the marker is a number do decode number otherwise forward to `deserialize_any`. 

If it is a number try convert the number into a i8; use num_traits if that conversion fails just fall back to visit_f64